### PR TITLE
Interface and its implementation

### PR DESCRIPTION
**Pull request by amitx007**
<hr>

## Description 📜
Explained about what is interface and how to implement it .
 # <2626> 

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] Java_Domain

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by amitx007** [comment]
@AkMo3 plz see the asset folder so that i can add the relative path to the doc.

**Comment by AkMo3** [comment]
@nandini7637 Please do the initial review

**Review by nandini7637 — CHANGES_REQUESTED** [review]
If you took references from any website please mention them. Else it's good!

**Comment by amitx007** [comment]
@nandini7637 no actually i have just seen the which version of java the extra features were added thats it .

**Comment by amitx007** [comment]
@nandini7637 now i have added all the relative paths to the documentation . So plz check it !

File: Java/Interfaces/Defining_And_Implementation_Of_Interfaces/Defining_and_implementation_of_interface.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+**NOTE :** **Loose coupling** in Java means classes that are independent of each other.
+
+-----------
+## Declaration of Interface 
+**SYNTAX** for declaring an interface 
+<!-- insert an image -->
+![Declarartion of interface](../Assets/interfaceDeclaration.png)
+
+here *smartphone* is the interface's name and *frontcamera*,*rearcamera* etc are the methods without the implemenatation (without body).
```
Comment (AkMo3) [inline Java/Interfaces/Defining_And_Implementation_Of_Interfaces/Defining_and_implementation_of_interface.md:20]: `implemenatation` -> `implementation`

File: Java/Interfaces/Defining_And_Implementation_Of_Interfaces/Defining_and_implementation_of_interface.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
+# Defining Interface and it's Implementation
```
Comment (AkMo3) [inline Java/Interfaces/Defining_And_Implementation_Of_Interfaces/Defining_and_implementation_of_interface.md:1]: `it's` -> `its`

File: Java/Interfaces/Defining_And_Implementation_Of_Interfaces/Defining_and_implementation_of_interface.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
+# Defining Interface and it's Implementation 
+---------
+## What is an Interface?
+An interface is quite similar to some other class, but it has got some method that have no body or in other words no implementation and some of the constant values. So in general interface is just a blueprint of a class which has static constant and abstract methods .
```
Comment (AkMo3) [inline Java/Interfaces/Defining_And_Implementation_Of_Interfaces/Defining_and_implementation_of_interface.md:4]: `other class` -> `other classes`
`some method` -> `some methods`
`class which has static constant ` -> `class that has static constants `

File: Java/Interfaces/Defining_And_Implementation_Of_Interfaces/Defining_and_implementation_of_interface.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+## Declaration of Interface 
+**SYNTAX** for declaring an interface 
+<!-- insert an image -->
+![Declarartion of interface](../Assets/interfaceDeclaration.png)
+
+here *smartphone* is the interface's name and *frontcamera*,*rearcamera* etc are the methods without the implemenatation (without body).
+
+------------
+## Methods and Variables in Interface 
+- All the **method** that are declared in interface are **public and abstract** by default .
```
Comment (AkMo3) [inline Java/Interfaces/Defining_And_Implementation_Of_Interfaces/Defining_and_implementation_of_interface.md:24]: `method` -> `methods`

**Review by AkMo3 — CHANGES_REQUESTED** [review]

**Comment by AkMo3** [comment]
@amitx007 Please remember to check the spell and grammar of your content through any online tool

**Comment by amitx007** [comment]
@AkMo3 checked all the grammar and spelling and writing convention ! plz do the review !

**Review by AkMo3 — APPROVED** [review]

**Pull request merged**